Test-Rack-Software/util/pio_commands.py

```diff
@@ -17,38 +21,55 @@
 #### Helper functions ####
 
 
+def run_script_threaded(arg_list, callback: callable = None):
+    """Runs a platformio command in a subthread. Calls `callback` every iteration."""
+    thr = threading.Thread(target=run_script, args=(arg_list,))
+    thr.start()
+    while thr.is_alive():
+        time.sleep(0.02)

```

Comment:
Comment why we need to sleep

Reply:
why exactly do we need to sleep?

Reply:
I think I copied it over from some other threading code, we should explore if this works as intended without the sleep.

My guess is I put it there to not inundate the `callback` with calls so I limited it to only 50 hz, but we should look into it.